context/src/repo/validator.rs

```diff
@@ -85,6 +86,10 @@ pub enum RepoValidationIssueSubOptimal {
 pub enum RepoValidationIssueInvalid {
     #[error("repo branch name too short")]
     RepoBranchNameTooShort(String),
+    #[error("repo sha too short")]
+    RepoShaTooShort(String),
+    #[error("repo sha too long, truncated to {}", MAX_SHA_FIELD_LEN)]
+    RepoShaTooLong(String),
```

Comment:
I think `RepoShaTooLong` should probably be a suboptimal error, especially if we're doing something like truncating it - things like `RepoAuthorNameTooLong` also are suboptimal errors. SHAs are mainly used as keys (meaning test results belonging to the same SHA have a high impact on whether or not a test is called flaky), so we care more about if they're there and less about their length (we care about them being too long I guess for data constraints??)